selfservice/strategy/code/strategy_recovery.go

```diff
@@ -0,0 +1,605 @@
+	"github.com/ory/kratos/x"
+)
+
+const (
+	RouteAdminCreateRecoveryCode = "/recovery/code"
+)
+
+func (s *Strategy) RecoveryStrategyID() string {
+	return recovery.StrategyRecoveryCodeName
+}
+
+func (s *Strategy) RegisterPublicRecoveryRoutes(public *x.RouterPublic) {
+	s.deps.CSRFHandler().IgnorePath(RouteAdminCreateRecoveryCode)
+	public.POST(RouteAdminCreateRecoveryCode, x.RedirectToAdminRoute(s.deps))
+
+}
+
+func (s *Strategy) RegisterAdminRecoveryRoutes(admin *x.RouterAdmin) {
+	wrappedCreateRecoveryCode := strategy.IsDisabled(s.deps, s.RecoveryStrategyID(), s.createRecoveryCode)
+	admin.POST(RouteAdminCreateRecoveryCode, wrappedCreateRecoveryCode)
+}
+
+func (s *Strategy) PopulateRecoveryMethod(r *http.Request, f *recovery.Flow) error {
+	f.UI.SetCSRF(s.deps.GenerateCSRFToken(r))
+	f.UI.GetNodes().Upsert(
+		node.NewInputField("email", nil, node.CodeGroup, node.InputAttributeTypeEmail, node.WithRequiredInputAttribute).
+			WithMetaLabel(text.NewInfoNodeInputEmail()),
+	)
+	f.UI.
+		GetNodes().
+		Append(node.NewInputField("method", s.RecoveryStrategyID(), node.CodeGroup, node.InputAttributeTypeSubmit).
+			WithMetaLabel(text.NewInfoNodeLabelSubmit()))
+
+	return nil
+}
+
+// swagger:parameters adminCreateSelfServiceRecoveryCode
+//
+// nolint
+type adminCreateSelfServiceRecoveryCode struct {
+	// in: body
+	Body adminCreateSelfServiceRecoveryCodeBody
+}
+
+// swagger:model adminCreateSelfServiceRecoveryCodeBody
+type adminCreateSelfServiceRecoveryCodeBody struct {
+	// Identity to Recover
+	//
+	// The identity's ID you wish to recover.
+	//
+	// required: true
+	IdentityID uuid.UUID `json:"identity_id"`
+
+	// Code Expires In
+	//
+	// The recovery code will expire at that point in time. Defaults to the configuration value of
+	// `selfservice.flows.recovery.request_lifespan`.
+	//
+	//
+	// pattern: ^[0-9]+(ns|us|ms|s|m|h)$
+	// example:
+	//	- 1h
+	//	- 1m
+	//	- 1s
+	ExpiresIn string `json:"expires_in"`
+}
+
+// swagger:model selfServiceRecoveryCode
+// nolint
+type selfServiceRecoveryCode struct {
+	// RecoveryLink with flow
+	//
+	// This link opens the recovery UI with an empty `code` field.
+	//
+	// required: true
+	// format: uri
+	RecoveryLink string `json:"recovery_link"`
+
+	// RecoveryCode is the code that can be used to recover the account
+	//
+	// required: true
+	RecoveryCode string `json:"recovery_code"`
+
+	// Expires At is the timestamp of when the recovery flow expires
+	//
+	// The timestamp when the recovery link expires.
+	ExpiresAt time.Time `json:"expires_at"`
+}
+
+// swagger:route POST /admin/recovery/code v0alpha2 adminCreateSelfServiceRecoveryCode
+//
+// # Create a Recovery Link
+//
+// This endpoint creates a recovery code which should be given to the user in order for them to recover
+// (or activate) their account.
+//
+//	Consumes:
+//	- application/json
+//
+//	Produces:
+//	- application/json
+//
+//	Schemes: http, https
+//
+//	Responses:
+//		200: selfServiceRecoveryCode
```

Comment:
As this creates the code, why not return 201?

Reply:
True, done.